### PR TITLE
more consistent work with paths in object storages

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Extracted from https://github.com/ClickHouse/ClickHouse/pull/36171.

### CODE REVIEW

File: src/Disks/ObjectStorages/DiskObjectStorageMetadata.h [hunk 1]
```diff
@@ -19,8 +19,8 @@ struct DiskObjectStorageMetadata
 
     const std::string & common_metadata_path;
 
-    /// Remote FS objects paths and their sizes.
-    std::vector<BlobPathWithSize> remote_fs_objects;
+    /// Relative paths of blobs.
+    std::vector<PathWithSize> storage_objects;
```
Comment (alesapin) [inline src/Disks/ObjectStorages/DiskObjectStorageMetadata.h:23]: Let's avoid confusion and use different structure here:
```
RelativePathWithSize
{
    path,
    size
}
```

File: src/Disks/ObjectStorages/DiskObjectStorageMetadata.cpp [hunk 2]
```diff
@@ -120,7 +120,7 @@ DiskObjectStorageMetadata::DiskObjectStorageMetadata(
 void DiskObjectStorageMetadata::addObject(const String & path, size_t size)
```
Comment (alesapin) [inline src/Disks/ObjectStorages/DiskObjectStorageMetadata.cpp:120]: need to check in some way that this path is relative.

**Review by alesapin — CHANGES_REQUESTED** [review]

File: src/Disks/ObjectStorages/DiskObjectStorageMetadata.h [hunk 3]
```diff
@@ -60,9 +60,9 @@ struct DiskObjectStorageMetadata
         return remote_fs_root_path;
     }
 
-    std::vector<BlobPathWithSize> getBlobs() const
+    std::vector<PathWithSize> getBlobs() const
```
Comment (alesapin) [inline src/Disks/ObjectStorages/DiskObjectStorageMetadata.h:63]: getBlobsRelativePaths

File: src/Disks/IO/ReadBufferFromWebServer.h [hunk 4]
```diff
@@ -19,7 +19,8 @@ class ReadBufferFromWebServer : public SeekableReadBuffer
 {
 public:
     explicit ReadBufferFromWebServer(
```
Comment (alesapin) [inline src/Disks/IO/ReadBufferFromWebServer.h:21]:
```suggestion
    ReadBufferFromWebServer(
```

**Review by alesapin — APPROVED** [review]

**Comment by excitoon** [comment]
Guys @kssenii @alesapin I wonder what is the purpose having sizes in `removeObjects`?

**Comment by kssenii** [comment]
> Guys @kssenii @alesapin I wonder what is the purpose having sizes in removeObjects?

Now there is no, but in another PR (https://github.com/ClickHouse/ClickHouse/pull/36171) it will be not just `PathsWithSize` but struct `StoredObject` and there it is needed.

**Pull request merged**